config.go

```diff
@@ -90,16 +22,22 @@ func parseConfiguration(configPath string) {
 	flag.Bool("prometheus.enabled", true, "Prometheus enabled")
 	flag.Int("http.port", 0, "http port")
 	flag.Int("grpc.port", 0, "grpc port")
+}
 
-	err := parsePropertyFileAndSetFlags(path.Join(conf, "application.properties"))
+func parseConfiguration(g *Gaz, configPath string) {
+	conf := GetConfigPath(configPath)
+
+	g.Viper.SetConfigName("application") //the suffix ".properties" will be added by viper
+	g.Viper.AddConfigPath(conf)
+	err := g.Viper.ReadInConfig()
 	if err != nil {
-		log.Fatalf("unable to read and extract key/value in %s: %v", conf+"/application.properties", err)
+		panic(err)
```

Comment:
Should we log something specific here in the panic?

Reply:
yes